src/main/java/hbv501g/recipes/Controllers/IngredientController.java

```diff
@@ -73,6 +80,49 @@ public Ingredient getIngredientByTitle(@PathVariable(value = "title") String tit
         return ingredientService.findByTitle(title);
     }
 
+    /**
+     * Endpoint createds new idgrediet for the database
+     *
+     * @param session : is the current session
+     * @param title : String value
+     * @param unit : is a Enum of unit
+     * @param quantity : double value
+     * @param price : double value
+     * @param store : String value
+     * @param brand : Stirng value
+     */
+    @RequestMapping("ingredient/created")
+    public Ingredient saveInredient(
+                    HttpSession session,
+				    @RequestParam String title,
+				    @RequestParam Unit unit,
+				    @RequestParam double quantity,
+				    @RequestParam double price,
+				    @RequestParam String store,
+				    @RequestParam String brand
+				    )
+    {
+	Ingredient ingredient = new Ingredient(title, unit, quantity, price, store, brand);
+    ingredient.setCreatedBy((User)session.getAttribute("LoggedInUser"));
+    ingredient.setDateOfCreation(LocalDate.now());
+	return ingredientService.save(ingredient);
+    }
+    
+    /**
+     * Endpoint that finds an ingredient by id and
+     * removes it form the database if the uesr
+     * own the ingredient.
+     * @param session : is the current session
+     * @param id      : ID number of the ingredient
+     */
+    @GetMapping("ingredient/delete/{id}")
+    public void deleteIngredientById(HttpSession session, @PathVariable(value = "id") long id)
+    {
+	if(ingredientService.findByID(id).getCreatedBy().getID() == ((User) session.getAttribute("LoggedInUser")).getID()){
```

Comment:
er ekki hægt að bera User saman hér, í staðin fyrir ID?

Reply:
nei það virkaði ekki, ég reindi og reindi en þetta var eina leið til að þetta væri hægt

Reply:
það er ekki hægt með == en ætti að vera hægt með .equals

Reply:
og þar sem id fyrir hvern huser er einstæty þá er þetta jarf gild